lib/src/view/game/game_player.dart

```diff
@@ -40,6 +41,10 @@ class GamePlayer extends StatelessWidget {
   });
 
   final Player player;
+
+  /// Whether we're displaying this player as part of an active game of the current user.
+  /// This might influence whether their rating is shown or not, based on the [ShowRatings] setting.
+  final bool isActiveGameOfCurrentUser;
```

Comment:
I'd rather not have this param and pass directly the `gameState` and the `side` instead of `player`.

Then you can infer everything from the the game state inside the `Player` widget. wdyt?

Reply:
Yeah that makes sense, will do 👍

Reply:
Ah it doesn't quite work though, since for the `archived_game_screen.dart` we only have a `LightArchivedGame` (instead of `ArchivedGame`) which does not have the `BaseGame` mixin that we would probably want to use here...?

Reply:
It's a bit annoying, because the `LightArchivedGame` still has all the info we need, but there's no common interface with `PlayableGame` - Maybe we could have something like `LightBaseGame`, that's like `BaseGame` but with less fields?

Reply:
I see the problem. A simple solution would be to delay the player display until we get the `ArchivedGame`, so you can still use the common interface without refactoring. I don't think it is much of an issue.